### PR TITLE
[yaml-front-matter] initial commit

### PR DESCRIPTION
**Pull request by xiaoxiangmoe**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by orta** [comment]
@xiaoxiangmoe Thank you for submitting this PR!  I see this is your first time submitting to DefinitelyTyped 👋 - keep an eye on this comment as I'll be updating it with information as things progress.


## Code Reviews

This PR adds a new definition, so it needs to be reviewed by a DT maintainer before it can be merged.

- [yaml-front-matter](https://www.npmjs.com/package/yaml-front-matter)


## Status

 * ✅ No merge conflicts
 * ❌ Continuous integration tests have passed
 * ❌ Only a DT maintainer can merge changes without tests

Once every item on this list is checked, I'll ask you for permission to merge and publish the changes.

----------------------
<details><summary>Diagnostic Information: What the bot saw about this PR</summary>

```json
{
  "type": "info",
  "pr_number": 43307,
  "author": "xiaoxiangmoe",
  "owners": [],
  "dangerLevel": "ScopedAndConfiguration",
  "headCommitAbbrOid": "c9ce891",
  "headCommitOid": "c9ce891ba6e5620fe35b424eeb4a436137bff711",
  "mergeIsRequested": false,
  "stalenessInDays": 0,
  "lastCommitDate": "2020-03-22T07:27:26.000Z",
  "reviewLink": "https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43307/files",
  "hasMergeConflict": false,
  "authorIsOwner": false,
  "isFirstContribution": true,
  "popularityLevel": "Well-liked by everyone",
  "anyPackageIsNew": true,
  "packages": [
    "yaml-front-matter"
  ],
  "files": [
    {
      "filePath": "types/yaml-front-matter/index.d.ts",
      "kind": "definition",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/tsconfig.json",
      "kind": "package-meta",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/tslint.json",
      "kind": "package-meta",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/yaml-front-matter-tests.ts",
      "kind": "test",
      "package": "yaml-front-matter"
    }
  ],
  "otherApprovalCount": 0,
  "ownerApprovalCount": 0,
  "maintainerApprovalCount": 0,
  "hasDismissedReview": false,
  "travisResult": "missing",
  "reviewersWithStaleReviews": [],
  "approvalFlags": 0,
  "isChangesRequested": false
}
```

</details>

**Comment by orta** [comment]
@xiaoxiangmoe Thank you for submitting this PR!  I see this is your first time submitting to DefinitelyTyped 👋 - keep an eye on this comment as I'll be updating it with information as things progress.


## Code Reviews

This PR adds a new definition, so it needs to be reviewed by a DT maintainer before it can be merged.

- [yaml-front-matter](https://www.npmjs.com/package/yaml-front-matter)


## Status

 * ✅ No merge conflicts
 * ❌ Continuous integration tests have passed
 * ❌ Only a DT maintainer can merge changes without tests

Once every item on this list is checked, I'll ask you for permission to merge and publish the changes.

----------------------
<details><summary>Diagnostic Information: What the bot saw about this PR</summary>

```json
{
  "type": "info",
  "pr_number": 43307,
  "author": "xiaoxiangmoe",
  "owners": [],
  "dangerLevel": "ScopedAndConfiguration",
  "headCommitAbbrOid": "c9ce891",
  "headCommitOid": "c9ce891ba6e5620fe35b424eeb4a436137bff711",
  "mergeIsRequested": false,
  "stalenessInDays": 0,
  "lastCommitDate": "2020-03-22T07:27:26.000Z",
  "reviewLink": "https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43307/files",
  "hasMergeConflict": false,
  "authorIsOwner": false,
  "isFirstContribution": true,
  "popularityLevel": "Well-liked by everyone",
  "anyPackageIsNew": true,
  "packages": [
    "yaml-front-matter"
  ],
  "files": [
    {
      "filePath": "types/yaml-front-matter/index.d.ts",
      "kind": "definition",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/tsconfig.json",
      "kind": "package-meta",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/tslint.json",
      "kind": "package-meta",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/yaml-front-matter-tests.ts",
      "kind": "test",
      "package": "yaml-front-matter"
    }
  ],
  "otherApprovalCount": 0,
  "ownerApprovalCount": 0,
  "maintainerApprovalCount": 0,
  "hasDismissedReview": false,
  "travisResult": "missing",
  "reviewersWithStaleReviews": [],
  "approvalFlags": 0,
  "isChangesRequested": false
}
```

</details>

**Comment by orta** [comment]
@xiaoxiangmoe Thank you for submitting this PR!  I see this is your first time submitting to DefinitelyTyped 👋 - keep an eye on this comment as I'll be updating it with information as things progress.


## Code Reviews

This PR adds a new definition, so it needs to be reviewed by a DT maintainer before it can be merged.

- [yaml-front-matter](https://www.npmjs.com/package/yaml-front-matter)


## Status

 * ✅ No merge conflicts
 * ❌ Continuous integration tests have passed
 * ❌ Only a DT maintainer can merge changes without tests

Once every item on this list is checked, I'll ask you for permission to merge and publish the changes.

----------------------
<details><summary>Diagnostic Information: What the bot saw about this PR</summary>

```json
{
  "type": "info",
  "pr_number": 43307,
  "author": "xiaoxiangmoe",
  "owners": [],
  "dangerLevel": "ScopedAndConfiguration",
  "headCommitAbbrOid": "c9ce891",
  "headCommitOid": "c9ce891ba6e5620fe35b424eeb4a436137bff711",
  "mergeIsRequested": false,
  "stalenessInDays": 0,
  "lastCommitDate": "2020-03-22T07:27:26.000Z",
  "reviewLink": "https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43307/files",
  "hasMergeConflict": false,
  "authorIsOwner": false,
  "isFirstContribution": true,
  "popularityLevel": "Well-liked by everyone",
  "anyPackageIsNew": true,
  "packages": [
    "yaml-front-matter"
  ],
  "files": [
    {
      "filePath": "types/yaml-front-matter/index.d.ts",
      "kind": "definition",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/tsconfig.json",
      "kind": "package-meta",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/tslint.json",
      "kind": "package-meta",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/yaml-front-matter-tests.ts",
      "kind": "test",
      "package": "yaml-front-matter"
    }
  ],
  "otherApprovalCount": 0,
  "ownerApprovalCount": 0,
  "maintainerApprovalCount": 0,
  "hasDismissedReview": false,
  "travisResult": "missing",
  "reviewersWithStaleReviews": [],
  "approvalFlags": 0,
  "isChangesRequested": false
}
```

</details>

**Comment by orta** [comment]
@xiaoxiangmoe Thank you for submitting this PR!  I see this is your first time submitting to DefinitelyTyped 👋 - keep an eye on this comment as I'll be updating it with information as things progress.


## Code Reviews

This PR adds a new definition, so it needs to be reviewed by a DT maintainer before it can be merged.

- [yaml-front-matter](https://www.npmjs.com/package/yaml-front-matter)


## Status

 * ✅ No merge conflicts
 * ✅ Continuous integration tests have passed
 * ❌ Only a DT maintainer can merge changes without tests

Once every item on this list is checked, I'll ask you for permission to merge and publish the changes.

----------------------
<details><summary>Diagnostic Information: What the bot saw about this PR</summary>

```json
{
  "type": "info",
  "pr_number": 43307,
  "author": "xiaoxiangmoe",
  "owners": [],
  "dangerLevel": "ScopedAndConfiguration",
  "headCommitAbbrOid": "c9ce891",
  "headCommitOid": "c9ce891ba6e5620fe35b424eeb4a436137bff711",
  "mergeIsRequested": false,
  "stalenessInDays": 0,
  "lastCommitDate": "2020-03-22T07:27:26.000Z",
  "reviewLink": "https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43307/files",
  "hasMergeConflict": false,
  "authorIsOwner": false,
  "isFirstContribution": true,
  "popularityLevel": "Well-liked by everyone",
  "anyPackageIsNew": true,
  "packages": [
    "yaml-front-matter"
  ],
  "files": [
    {
      "filePath": "types/yaml-front-matter/index.d.ts",
      "kind": "definition",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/tsconfig.json",
      "kind": "package-meta",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/tslint.json",
      "kind": "package-meta",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/yaml-front-matter-tests.ts",
      "kind": "test",
      "package": "yaml-front-matter"
    }
  ],
  "otherApprovalCount": 0,
  "ownerApprovalCount": 0,
  "maintainerApprovalCount": 0,
  "hasDismissedReview": false,
  "travisResult": "pass",
  "reviewersWithStaleReviews": [],
  "approvalFlags": 0,
  "isChangesRequested": false
}
```

</details>

**Comment by orta** [comment]
@xiaoxiangmoe Thank you for submitting this PR!  I see this is your first time submitting to DefinitelyTyped 👋 - keep an eye on this comment as I'll be updating it with information as things progress.


## Code Reviews

This PR adds a new definition, so it needs to be reviewed by a DT maintainer before it can be merged.

- [yaml-front-matter](https://www.npmjs.com/package/yaml-front-matter)


## Status

 * ✅ No merge conflicts
 * ✅ Continuous integration tests have passed
 * ❌ Only a DT maintainer can merge changes without tests

Once every item on this list is checked, I'll ask you for permission to merge and publish the changes.

----------------------
<details><summary>Diagnostic Information: What the bot saw about this PR</summary>

```json
{
  "type": "info",
  "pr_number": 43307,
  "author": "xiaoxiangmoe",
  "owners": [],
  "dangerLevel": "ScopedAndConfiguration",
  "headCommitAbbrOid": "c9ce891",
  "headCommitOid": "c9ce891ba6e5620fe35b424eeb4a436137bff711",
  "mergeIsRequested": false,
  "stalenessInDays": 0,
  "lastCommitDate": "2020-03-22T07:27:26.000Z",
  "reviewLink": "https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43307/files",
  "hasMergeConflict": false,
  "authorIsOwner": false,
  "isFirstContribution": true,
  "popularityLevel": "Well-liked by everyone",
  "anyPackageIsNew": true,
  "packages": [
    "yaml-front-matter"
  ],
  "files": [
    {
      "filePath": "types/yaml-front-matter/index.d.ts",
      "kind": "definition",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/tsconfig.json",
      "kind": "package-meta",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/tslint.json",
      "kind": "package-meta",
      "package": "yaml-front-matter"
    },
    {
      "filePath": "types/yaml-front-matter/yaml-front-matter-tests.ts",
      "kind": "test",
      "package": "yaml-front-matter"
    }
  ],
  "otherApprovalCount": 0,
  "ownerApprovalCount": 0,
  "maintainerApprovalCount": 0,
  "hasDismissedReview": false,
  "travisResult": "pass",
  "reviewersWithStaleReviews": [],
  "approvalFlags": 0,
  "isChangesRequested": false
}
```

</details>

**Comment by xiaoxiangmoe** [comment]
Reviewers: No reviews.

What should I do? @orta

**Review by orta — APPROVED** [review]

**Comment by orta** [comment]
Looks good to me, thanks

**Pull request merged**
